File: frontend/src/lib/statuts.ts
/**
 * Libellés et variantes visuelles de chaque statut backend — une seule source.
 * Les valeurs (snake_case) sont celles de la charte API, jamais renommées.
 */
export type VarianteStatut = 'neutre' | 'info' | 'gain' | 'perte' | 'alerte' | 'volt'

export interface DescriptionStatut {
  libelle: string
  variante: VarianteStatut
  /** Statut « vivant » : le badge pulse lentement. */
  actif?: boolean
}

const DEFIS: Record<string, DescriptionStatut> = {
  ouvert: { libelle: 'Ouvert', variante: 'volt', actif: true },
  complet: { libelle: 'Complet', variante: 'info' },
  annule: { libelle: 'Annulé', variante: 'neutre' },
  expire: { libelle: 'Expiré', variante: 'neutre' },
}

/**
 * Statuts de match. `verification` ne fait plus partie du parcours joueur (deux déclarations
 * concordantes règlent le match immédiatement) mais reste décrit ici : des lignes historiques
 * y sont encore et l'administration peut toujours les régler à la main.
 */
const MATCHS: Record<string, DescriptionStatut> = {
  en_cours: { libelle: 'En cours', variante: 'info', actif: true },
  preuve_requise: { libelle: 'Preuve exigée', variante: 'alerte', actif: true },
  nul_en_attente: { libelle: 'Match nul', variante: 'alerte', actif: true },
  verification: { libelle: 'Vérification', variante: 'alerte', actif: true },
  litige: { libelle: 'Litige', variante: 'perte', actif: true },
  termine: { libelle: 'Terminé', variante: 'gain' },
}

const LITIGES: Record<string, DescriptionStatut> = {
  en_cours: { libelle: 'En arbitrage', variante: 'alerte', actif: true },
  resolu: { libelle: 'Résolu', variante: 'gain' },
}

const PREUVES: Record<string, DescriptionStatut> = {
  en_attente: { libelle: 'En attente', variante: 'alerte' },
  validee: { libelle: 'Validée', variante: 'gain' },
  rejetee: { libelle: 'Rejetée', variante: 'perte' },
}

const PAIEMENTS: Record<string, DescriptionStatut> = {
  en_attente: { libelle: 'En attente', variante: 'alerte', actif: true },
  reussi: { libelle: 'Réussi', variante: 'gain' },
  echoue: { libelle: 'Échoué', variante: 'perte' },
  rembourse: { libelle: 'Remboursé', variante: 'info' },
}

const TRANSACTIONS: Record<string, DescriptionStatut> = {
  valide: { libelle: 'Validé', variante: 'gain' },
  en_attente: { libelle: 'En attente', variante: 'alerte' },
  annule: { libelle: 'Annulé', variante: 'neutre' },
}

const UTILISATEURS: Record<string, DescriptionStatut> = {
  actif: { libelle: 'Actif', variante: 'gain' },
  suspendu: { libelle: 'Suspendu', variante: 'perte' },
  en_attente: { libelle: 'En attente', variante: 'alerte' },
  supprime: { libelle: 'Supprimé', variante: 'neutre' },
}

const CATALOGUE: Record<string, DescriptionStatut> = {
  actif: { libelle: 'Actif', variante: 'gain' },
  inactif: { libelle: 'Inactif', variante: 'neutre' },
}

const MISES: Record<string, DescriptionStatut> = {
  bloquee: { libelle: 'Bloquée', variante: 'alerte' },
  gagnee: { libelle: 'Gagnée', variante: 'gain' },
  perdue: { libelle: 'Perdue', variante: 'perte' },
  remboursee: { libelle: 'Remboursée', variante: 'info' },
}

/** Messages du formulaire de contact (module `contact`). */
const CONTACT: Record<string, DescriptionStatut> = {
  nouveau: { libelle: 'Nouveau', variante: 'volt', actif: true },
  lu: { libelle: 'Lu', variante: 'info' },
  traite: { libelle: 'Traité', variante: 'gain' },
}

export const statuts = {
  defi: DEFIS,
  match: MATCHS,
  litige: LITIGES,
  preuve: PREUVES,
  paiement: PAIEMENTS,
  transaction: TRANSACTIONS,
  utilisateur: UTILISATEURS,
  catalogue: CATALOGUE,
  mise: MISES,
  contact: CONTACT,
} as const

export type FamilleStatut = keyof typeof statuts

export function decrireStatut(famille: FamilleStatut, valeur: string): DescriptionStatut {
  return statuts[famille][valeur] ?? { libelle: valeur.replaceAll('_', ' '), variante: 'neutre' }
}

/** Libellés des types de mouvement du portefeuille. */
export const typesTransaction: Record<string, { libelle: string; sens: 'credit' | 'debit' | 'neutre' }> = {
  depot: { libelle: 'Dépôt', sens: 'credit' },
  mise_bloquee: { libelle: 'Mise bloquée', sens: 'debit' },
  gain: { libelle: 'Gain de match', sens: 'credit' },
  commission: { libelle: 'Commission', sens: 'neutre' },
  remboursement: { libelle: 'Remboursement', sens: 'credit' },
  retrait: { libelle: 'Retrait', sens: 'debit' },
}

export const typesNotification: Record<string, string> = {
  defi_rejoint: 'Défi accepté',
  defi_expire: 'Défi expiré',
  match_a_valider: 'Match à valider',
  // Nouveau parcours de fin de match (confirmation, désaccord, nul, rejeu, abandon).
  match_score: 'Résultat à confirmer',
  match_desaccord: 'Déclarations divergentes',
  match_nul: 'Match nul',
  match_rejoue: 'Nouvelle manche',
  match_abandon: 'Délai de confirmation écoulé',
  match_termine: 'Match terminé',
  litige_ouvert: 'Litige ouvert',
  litige_resolu: 'Litige résolu',
  paiement_confirme: 'Paiement confirmé',
  paiement_echoue: 'Paiement échoué',
}

export const libellesPrestataires: Record<string, string> = {
  ligdicash: 'LigdiCash',
  fusionmoney: 'MoneyFusion',
}

export const libellesConfigurations: Record<string, { libelle: string; unite: 'pourcentage' | 'montant'; aide: string }> = {
  commission_defi: {
    libelle: 'Commission par défi',
    unite: 'pourcentage',
    aide: 'Prélevée sur le total des deux mises au règlement du match.',
  },
  mise_minimale: { libelle: 'Mise minimale', unite: 'montant', aide: 'Montant plancher par joueur pour créer un défi.' },
  mise_maximale: { libelle: 'Mise maximale', unite: 'montant', aide: 'Montant plafond par joueur pour créer un défi.' },
  frais_retrait: {
    libelle: 'Frais de retrait',
    unite: 'pourcentage',
    aide: 'Prélevés en plus du montant retiré, acquis seulement si le retrait réussit.',
  },
}

export const libellesActionsAudit: Record<string, string> = {
  'defi:creation': 'Création de défi',
  'defi:rejoindre': 'Défi rejoint',
  'defi:annulation': 'Défi annulé',
  'defi:expiration': 'Défi expiré',
  'match:validation': 'Match validé',
  'litige:ouverture': 'Litige ouvert',
  'litige:decision': 'Décision de litige',
  'litige:relance': 'Relance arbitre',
  'preuve:validee': 'Preuve validée',
  'preuve:rejetee': 'Preuve rejetée',
  'paiement:depot_reussi': 'Dépôt crédité',
  'paiement:retrait_demande': 'Retrait demandé',
  'paiement:statut_reussi': 'Paiement réussi',
  'paiement:statut_echoue': 'Paiement échoué',
  'paiement:statut_rembourse': 'Paiement remboursé',
  'utilisateur:statut_actif': 'Compte réactivé',
  'utilisateur:statut_suspendu': 'Compte suspendu',
  'utilisateur:creation': 'Compte créé par l’admin',
  'utilisateur:modification': 'Compte modifié par l’admin',
  'utilisateur:suppression': 'Compte supprimé (anonymisé)',
  'contact:statut_nouveau': 'Message remis à « nouveau »',
  'contact:statut_lu': 'Message marqué lu',
  'contact:statut_traite': 'Message marqué traité',
  'contact:suppression': 'Message de contact supprimé',
  'configuration:modification': 'Configuration modifiée',
  'connexion:echec_joueur': 'Échec de connexion joueur',
  'connexion:echec_admin': 'Échec de connexion admin',
}

File: mobile/lib/noyau/statuts.dart
import 'package:flutter/material.dart';

import '../theme/couleurs.dart';

/// Libellés et couleurs de chaque statut du backend — miroir de
/// `frontend/src/lib/statuts.ts`. Les valeurs (`snake_case`) sont celles de la
/// charte API et ne sont JAMAIS renommées côté mobile.

class DescriptionStatut {
  const DescriptionStatut(this.libelle, this.texte, this.fond, {this.actif = false});

  final String libelle;
  final Color texte;
  final Color fond;

  /// Statut « vivant » : le badge pulse lentement.
  final bool actif;
}

const Map<String, DescriptionStatut> statutsDefi = {
  'ouvert': DescriptionStatut('Ouvert', Couleurs.vert, Couleurs.voltFond, actif: true),
  'complet': DescriptionStatut('Complet', Couleurs.info, Couleurs.infoFond),
  'annule': DescriptionStatut('Annulé', Couleurs.muet, Couleurs.gris),
  'expire': DescriptionStatut('Expiré', Couleurs.muet, Couleurs.gris),
};

/// `verification` ne fait plus partie du parcours joueur (deux déclarations
/// concordantes règlent le match immédiatement) mais reste décrit : des lignes
/// historiques y sont encore et l'administration peut les régler à la main.
const Map<String, DescriptionStatut> statutsMatch = {
  'en_cours': DescriptionStatut('En cours', Couleurs.info, Couleurs.infoFond, actif: true),
  'preuve_requise':
      DescriptionStatut('Preuve exigée', Couleurs.alerte, Couleurs.alerteFond, actif: true),
  'nul_en_attente':
      DescriptionStatut('Match nul', Couleurs.alerte, Couleurs.alerteFond, actif: true),
  'verification':
      DescriptionStatut('Vérification', Couleurs.alerte, Couleurs.alerteFond, actif: true),
  'litige': DescriptionStatut('Litige', Couleurs.perte, Couleurs.perteFond, actif: true),
  'termine': DescriptionStatut('Terminé', Couleurs.gain, Couleurs.gainFond),
};

const Map<String, DescriptionStatut> statutsLitige = {
  'en_cours':
      DescriptionStatut('En arbitrage', Couleurs.alerte, Couleurs.alerteFond, actif: true),
  'resolu': DescriptionStatut('Résolu', Couleurs.gain, Couleurs.gainFond),
};

const Map<String, DescriptionStatut> statutsPreuve = {
  'en_attente': DescriptionStatut('En attente', Couleurs.alerte, Couleurs.alerteFond),
  'validee': DescriptionStatut('Validée', Couleurs.gain, Couleurs.gainFond),
  'rejetee': DescriptionStatut('Rejetée', Couleurs.perte, Couleurs.perteFond),
};

const Map<String, DescriptionStatut> statutsPaiement = {
  'en_attente':
      DescriptionStatut('En attente', Couleurs.alerte, Couleurs.alerteFond, actif: true),
  'reussi': DescriptionStatut('Réussi', Couleurs.gain, Couleurs.gainFond),
  'echoue': DescriptionStatut('Échoué', Couleurs.perte, Couleurs.perteFond),
  'rembourse': DescriptionStatut('Remboursé', Couleurs.info, Couleurs.infoFond),
};

const Map<String, DescriptionStatut> statutsTransaction = {
  'valide': DescriptionStatut('Validé', Couleurs.gain, Couleurs.gainFond),
  'en_attente': DescriptionStatut('En attente', Couleurs.alerte, Couleurs.alerteFond),
  'annule': DescriptionStatut('Annulé', Couleurs.muet, Couleurs.gris),
};

enum FamilleStatut { defi, match, litige, preuve, paiement, transaction }

DescriptionStatut decrireStatut(FamilleStatut famille, String valeur) {
  final table = switch (famille) {
    FamilleStatut.defi => statutsDefi,
    FamilleStatut.match => statutsMatch,
    FamilleStatut.litige => statutsLitige,
    FamilleStatut.preuve => statutsPreuve,
    FamilleStatut.paiement => statutsPaiement,
    FamilleStatut.transaction => statutsTransaction,
  };
  return table[valeur] ??
      DescriptionStatut(valeur.replaceAll('_', ' '), Couleurs.muet, Couleurs.gris);
}

/// Type de mouvement du portefeuille : libellé et sens (crédit / débit).
class TypeTransaction {
  const TypeTransaction(this.libelle, this.sens);
  final String libelle;
  final String sens; // credit | debit | neutre
}

const Map<String, TypeTransaction> typesTransaction = {
  'depot': TypeTransaction('Dépôt', 'credit'),
  'mise_bloquee': TypeTransaction('Mise bloquée', 'debit'),
  'gain': TypeTransaction('Gain de match', 'credit'),
  'commission': TypeTransaction('Commission', 'neutre'),
  'remboursement': TypeTransaction('Remboursement', 'credit'),
  'retrait': TypeTransaction('Retrait', 'debit'),
};

const Map<String, String> typesNotification = {
  'defi_rejoint': 'Défi accepté',
  'defi_expire': 'Défi expiré',
  'match_a_valider': 'Match à valider',
  'match_score': 'Score à confirmer',
  'match_desaccord': 'Déclarations divergentes',
  'match_nul': 'Match nul',
  'match_rejoue': 'Nouvelle manche',
  'match_abandon': 'Délai de confirmation écoulé',
  'match_termine': 'Match terminé',
  'litige_ouvert': 'Litige ouvert',
  'litige_resolu': 'Litige résolu',
  'paiement_confirme': 'Paiement confirmé',
  'paiement_echoue': 'Paiement échoué',
};

/// Icône associée à chaque type de notification (barre de la liste).
const Map<String, IconData> iconesNotification = {
  'defi_rejoint': Icons.handshake_outlined,
  'defi_expire': Icons.schedule,
  'match_a_valider': Icons.sports_esports_outlined,
  'match_score': Icons.scoreboard_outlined,
  'match_desaccord': Icons.report_problem_outlined,
  'match_nul': Icons.balance,
  'match_rejoue': Icons.replay,
  'match_abandon': Icons.timer_off_outlined,
  'match_termine': Icons.emoji_events_outlined,
  'litige_ouvert': Icons.gavel_outlined,
  'litige_resolu': Icons.gavel,
  'paiement_confirme': Icons.south_west,
  'paiement_echoue': Icons.error_outline,
};

